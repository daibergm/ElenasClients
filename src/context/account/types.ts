export enum ActionTypes {
  LOGIN_ATTEMPT = '@ACCOUNT_LOGIN_ATTEMPT',
  LOGIN_SUCCESS = '@ACCOUNT_LOGIN_SUCCESS',
  LOGIN_FAILURE = '@ACCOUNT_LOGIN_FAILURE',
  LOGOUT = '@ACCOUNT_LOGOUT',
}

export type Action = {
  type:
    | ActionTypes.LOGIN_ATTEMPT
    | ActionTypes.LOGIN_SUCCESS
    | ActionTypes.LOGIN_FAILURE
    | ActionTypes.LOGOUT;
  payload?: string;
};
