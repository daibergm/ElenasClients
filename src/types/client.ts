export type Client = {
  id?: number;
  firstName: string;
  lastName: string;
  cedula?: string;
  email?: string;
  cellphone?: string;
  address: string;
};
