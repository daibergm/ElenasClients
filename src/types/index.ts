export * from './routes';
export * from './user';
export * from './client';
