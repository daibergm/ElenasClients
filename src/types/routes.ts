import { RouteProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';

export type ClientsStackParamList = {
  LOGIN: undefined;
  CLIENTS: undefined;
  CLIENT: { clientId?: number };
};

export type ClientNavigationProps = StackNavigationProp<
  ClientsStackParamList,
  'CLIENT'
>;

export type ClientRouteProps = RouteProp<ClientsStackParamList, 'CLIENT'>;
