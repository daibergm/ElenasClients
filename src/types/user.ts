export type Login = {
  cellphone: string;
  password: string;
};

export type User = {
  cellphone: string;
  firstName: string;
  lastName: string;
  DNI: string;
  email: string;
};
