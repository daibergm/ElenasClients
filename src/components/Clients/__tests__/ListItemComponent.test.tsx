import React from 'react';
import { render } from '@testing-library/react-native';

// @Components
import ListItem from '../ListItemComponent';

// @Types
import { Client } from '../../../types';

describe('ListItem test suits', () => {
  const item: Client = {
    id: 1,
    firstName: 'John',
    lastName: 'Doe',
    cellphone: '123',
  };

  it('Render correctly', () => {
    const onPress = jest.fn();
    const listItem = render(<ListItem item={item} onPress={onPress} />);
    expect(listItem).toMatchSnapshot();
  });
  it('Test onPress', () => {
    const onPress = jest.fn();
    const listItem = render(<ListItem item={item} onPress={onPress} />);
    listItem.toJSON()?.props.onClick();
    expect(onPress).toBeCalled();
  });
});
