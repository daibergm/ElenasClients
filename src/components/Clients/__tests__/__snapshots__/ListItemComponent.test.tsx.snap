// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem test suits Render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 1,
      "marginBottom": 10,
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text>
    John Doe
  </Text>
  <Text>
    Celular: 123
  </Text>
</View>
`;
